server/src/main/java/com/linecorp/centraldogma/server/internal/api/sysadmin/ApplicationRegistryService.java

```diff
@@ -0,0 +1,462 @@
+            }
+        } else {
+            checkArgument(certificateId != null, "CERTIFICATE type must have a certificateId.");
+            checkArgument(secret == null,
+                          "CERTIFICATE type cannot have a secret: %s", secret);
+            future = mds.createCertificate(author, appId, certificateId, isSystemAdmin);
+        }
+        return future.thenCompose(unused -> fetchApplicationByAppId(appId))
+                     .thenApply(application -> {
+                         final ResponseHeaders headers = ResponseHeaders.of(HttpStatus.CREATED,
+                                                                            HttpHeaderNames.LOCATION,
+                                                                            "/applications/" + appId);
+                         return ResponseEntity.of(headers, application);
+                     });
+    }
+
+    /**
+     * DELETE /applications/{appId}
+     *
+     * <p>Deletes an application of the specified ID then returns it.
+     */
+    @Delete("/applications/{appId}")
+    public CompletableFuture<Application> deleteApplication(ServiceRequestContext ctx,
+                                                            @Param String appId,
+                                                            Author author, User loginUser) {
+        return getApplicationOrRespondForbidden(ctx, appId, loginUser).thenCompose(
+                application -> {
+                    if (application.type() == ApplicationType.TOKEN) {
+                        return mds.destroyToken(author, appId)
+                                  .thenApply(unused -> ((Token) application).withoutSecret());
+                    }
+                    return mds.destroyCertificate(author, appId).thenApply(unused -> application);
+                });
+    }
+
+    /**
+     * DELETE /applications/{appId}/removed
+     *
+     * <p>Purges an application of the specified ID that was deleted before.
+     */
+    @Delete("/applications/{appId}/removed")
+    @RequiresSystemAdministrator
+    public CompletableFuture<Application> purgeApplication(ServiceRequestContext ctx,
+                                                           @Param String appId,
+                                                           Author author, User loginUser) {
+        return getApplicationOrRespondForbidden(ctx, appId, loginUser).thenApplyAsync(
+                application -> {
+                    mds.purgeApplication(author, appId);
+                    if (application.type() == ApplicationType.TOKEN) {
+                        return ((Token) application).withoutSecret();
+                    }
+                    return application;
+                }, ctx.blockingTaskExecutor());
+    }
+
+    /**
+     * PATCH /applications/{appId}
+     *
+     * <p>Activates or deactivates the application of the specified {@code appId}.
+     */
+    @Patch("/applications/{appId}")
+    @Consumes("application/json-patch+json")
```

Comment:
Does it now consume json?
```suggestion
    @Consumes("application/json")
```